src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy

```diff
@@ -193,24 +199,34 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
 
       boolean needErrorProneJavac = javaVersion.java8 && isJavac9CF
 
+      def createManifestTask = project.task('createCheckerFrameworkManifest', type: CreateManifestTask)
+      createManifestTask.checkers = userConfig.checkers
 
       project.tasks.withType(AbstractCompile).all { compile ->
         if (compile.hasProperty('options') && (!userConfig.excludeTests || !compile.name.toLowerCase().contains("test"))) {
           compile.options.annotationProcessorPath =
                   compile.options.annotationProcessorPath == null ?
                           project.configurations.checkerFramework :
                           project.configurations.checkerFramework.plus(compile.options.annotationProcessorPath)
-            // Check whether to use the Error Prone javac
-            if (needErrorProneJavac) {
+          // Check whether to use the Error Prone javac
+          if (needErrorProneJavac) {
             compile.options.forkOptions.jvmArgs += [
               "-Xbootclasspath/p:${project.configurations.errorProneJavac.asPath}".toString()
             ]
           }
-          compile.options.compilerArgs = [
+          compile.options.compilerArgs += [
             "-Xbootclasspath/p:${project.configurations.checkerFrameworkAnnotatedJDK.asPath}".toString()
           ]
           if (!userConfig.checkers.empty) {
-            compile.options.compilerArgs << "-processor" << userConfig.checkers.join(",")
+            compile.dependsOn(createManifestTask)
+            // Add the manifest file to the annotation processor path, so that the javac
+            // annotation processor discovery mechanism finds the checkers to use and
+            // runs them alongside any other annotation processors on the processor path.
+            // Using the -processor flag to specify checkers would make the plugin incompatible
+            // with other annotation processors, which are typically auto-discovered by javac.
```

Comment:
"typically" may be true, but it is not relevant.  Just "annotation processors that are auto-discovered" is shorter and keeps the focus where it belongs.